src/resources/elements/NetworkFeedback/NetworkFeedback.ts

```diff
@@ -20,15 +20,21 @@ export class NetworkFeedback {
     this.network = EthereumService.targetedNetwork;
     this.show = false;
 
-    this.isProduction = process.env.NODE_ENV !== "development" || process.env.NETWORK === "mainnet";
-    const storedNetwork = this.storageService.lsGet<AllowedNetworks>("network");
-    const isMainnet = [Networks.Mainnet, Networks.Celo, Networks.Arbitrum].includes(storedNetwork);
-    const isTestnet = [Networks.Rinkeby, Networks.Alfajores, Networks.Kovan].includes(storedNetwork);
-    if (
-      storedNetwork &&
-      (this.isProduction && !isMainnet) || (!this.isProduction && !isTestnet)
-    ) {
-      this.storageService.lsRemove("network");
+    this.isProductionEnv = process.env.NODE_ENV !== "development" || process.env.NETWORK === Networks.Mainnet;
+    const locallyStoredNetwork = this.storageService.lsGet<AllowedNetworks>("network");
+    if (locallyStoredNetwork) {
+      const defaultNetwork = this.isProductionEnv ? Networks.Mainnet : Networks.Rinkeby;
```

Comment:
future: put variable closer to where it is used.
why was it put before the `if (illegalNetwork) {` block?